src/lib/Microsoft.Fx.Portability.MetadataReader/MetadataReaderExtensions.cs

```diff
@@ -93,52 +93,40 @@ private static AssemblyReferenceInformation FormatAssemblyInfo(this MetadataRead
                 ? "neutral"
                 : metadataReader.GetString(cultureHandle);
 
-            var publicKeyToken = publicKeyTokenHandle.IsNil
-                ? "null"
-                : metadataReader.FormatPublicKeyToken(publicKeyTokenHandle);
-
-            return new AssemblyReferenceInformation(name, version, culture, publicKeyToken);
+            return new AssemblyReferenceInformation(name, version, culture, metadataReader.GetPublicKeyToken(publicKeyTokenHandle));
         }
 
-        /// <summary>
-        /// Convert a blob referencing a public key token from a PE file into a human-readable string.
-        ///
-        /// If there are no bytes, the return will be 'null'
-        /// If the length is greater than 8, it is a strong name signed assembly
-        /// Otherwise, the key is the byte sequence.
-        /// </summary>
-        [SuppressMessage("Microsoft.Security.Cryptography", "CA5354:SHA1CannotBeUsed", Justification = "Public key tokens are calculated using a SHA-1 hash.")]
-        private static string FormatPublicKeyToken(this MetadataReader metadataReader, BlobHandle handle)
+        public static PublicKeyToken GetPublicKeyToken(this MetadataReader metadataReader, BlobHandle handle)
         {
-            byte[] bytes = metadataReader.GetBlobBytes(handle);
-
-            if (bytes == null || bytes.Length <= 0)
+            if (handle.IsNil)
             {
-                return "null";
+                return default;
             }
 
+            var bytes = metadataReader.GetBlobBytes(handle);
+
             // Strong named assembly
-            if (bytes.Length > 8)
+            if (bytes != null && bytes.Length > 8)
```

Comment:
Not sure what you mean. This method is to create the public key token from a blob handle; not sure how to separate anything out without losing clear ownership of `bytes`.

Reply:
Your yesterday's clean up commit with "return new PublicKeyToken(ImmutableArray.Create(bytes));" addressed my concern.

Reply:
Ah - that's because I had an implicit conversion from immutablearray<byte> to PublicKeyToken but decided to make it more explicit. Sounds like that was a good choice :)